client/state/data-layer/wpcom/sites/jitm/index.js

```diff
@@ -0,0 +1,175 @@
+		apiNamespace: 'rest',
+		method: 'GET',
+		path: `/v1.1/jetpack-blogs/${ currentSite }/rest-api/`,
+		query: {
+			path: '/jetpack/v4/jitm',
+			query: JSON.stringify( { message_path: `calypso:${ process.lastSection }:admin_notices` } ),
+			http_envelope: 1
+		}
+	}, { ...action, messagePath: process.lastSection } ) );
+};
+
+/**
+ * Called when a route change might have occured
+ * @param {function} getState A function to retrieve the current state
+ * @param {string} action The action being processed
+ * @param {function} dispatch A function to dispatch an action
+ */
+export const handleRouteChange = ( { getState, dispatch }, action ) => {
+	if ( ! config.isEnabled( 'jitms' ) ) {
+		return;
+	}
+
+	if ( process.hasInitializedSection && action.section && process.lastSection === action.section.name ) {
+		return;
+	}
+
+	// wait for loading to complete before trying to show a jitm since we cannot capture enough state until
+	// loading completes
+	switch ( action.isLoading ) {
+		case false:
+			process.hasInitializedSection = true;
+			return;
+		case true:
+			process.hasInitializedSection = false;
+			return;
+	}
+
+	process.lastSection = action.section.name;
+
+	fetchJITM( getState(), dispatch, action );
+};
+
+/**
+ * Called when a site is selected
+ * @param {function} getState Function to get the current state
+ * @param {string} action The action being processed
+ * @param {function} dispatch The dispatch function
+ */
+export const handleSiteSelection = ( { getState, dispatch }, action ) => {
+	if ( ! config.isEnabled( 'jitms' ) ) {
+		return;
+	}
+
+	if ( process.hasInitializedSites && process.lastSite === action.siteId ) {
+		return;
+	}
+
+	process.hasInitializedSites = ! ! action.siteId;
+	process.lastSite = action.siteId;
+
+	fetchJITM( getState(), dispatch, action );
+};
+
+/**
+ * Called when the http layer receives a valid jitm
+ * @param {function} dispatch The dispatch function
+ * @param {number} siteId The site id
+ * @param {object} jitms The jitms
+ * @param {number} site_id The site id
+ * @return {undefined} Nothing
+ */
+export const receiveJITM = ( { dispatch }, { siteId, site_id, messagePath }, jitms ) =>
+	insertJITM( dispatch, siteId || site_id, messagePath, jitms );
+
+/**
+ * Called when a jitm fails for any network related reason
+ * @param {function} dispatch The dispatch function
+ * @param {number} siteId The site id
+ * @param {number} site_id The site id
+ * @return {undefined} Nothing
+ */
+export const failedJITM = ( { dispatch }, { siteId, site_id, messagePath } ) =>
+	insertJITM( dispatch, siteId || site_id, messagePath, [] );
+
+export default {
+	[ SECTION_SET ]: [
+		dispatchRequest(
+			handleRouteChange,
+			receiveJITM,
+			failedJITM,
+			{ fromApi: makeParser( schema, {}, transformApiRequest ) }
+		)
+	],
+	[ SELECTED_SITE_SET ]: [
+		dispatchRequest(
+			handleSiteSelection,
+			receiveJITM,
+			failedJITM,
+			{ fromApi: makeParser( schema, {}, transformApiRequest ) }
+		)
+	],
+};
```

Comment:
if we move the `config.isEnabled()` call down here then we can deal with it once only, better yet - move it up to the parent where we merge this in to the other handlers.